### ISSUE TITLE
unable to use ApiQuery with a $ref schema

### ISSUE THREAD

**Issue by PandaaAgency** [nestjs/swagger#1013]
## I'm submitting a...

<pre><code>
[ ] Regression 
[x] Bug report
[ ] Feature request
[ ] Documentation issue or request
[ ] Support request => Please do not submit support request here, instead post your question on Stack Overflow.
</code></pre>

## Current behavior

Using  `ApiQuery ` decorator, i'm unable to make schema using `$ref`

```
@ApiQuery({
    required: false,
    name: nested.field,
    style: 'deepObject',
    schema: {
        $ref: getSchemaPath(MyAwesomeDto),
    },
}),
```
> Type '{ $ref: string; }' is not assignable to type 'SchemaObject'.
  Object literal may only specify known properties, and '$ref' does not exist in type 'SchemaObject'

i'm actually doing a dirty workaround setting a `default`  with a value starting with `$ref/` and updating swagger document withing my `main.ts`.

It seems to be a type issue fromhttps://github.com/nestjs/swagger/blob/master/lib/decorators/api-query.decorator.ts#L26 . 
I didn't test but shouldn't this line be : 
`  schema: SchemaObject | ReferenceObject `

## Expected behavior

Having the possibility to create an `ApiQuery ` with all parameters available

## Environment

```
"@nestjs/core": "^7.4.4",
"@nestjs/swagger": "^4.6.1",
```

Thanks everyone,

**Comment by kamilmysliwiec** [comment]
Would you like to create a PR for this issue?

**Comment by kamilmysliwiec** [comment]
This should be fixed in 4.7.3
